src/dataServices/dynamoDbBundleService.ts

```diff
@@ -270,6 +286,81 @@ export class DynamoDbBundleService implements Bundle {
         }
     }
 
+    private async updatedReferences(requests: BatchReadWriteRequest[], lockedItems: ItemRequest[]) {
+        const idToVersionId: Record<string, string> = {};
+        lockedItems.forEach((itemRequest: ItemRequest) => {
+            if (itemRequest.operation === 'update' && itemRequest.vid) {
+                idToVersionId[`${itemRequest.resourceType}_${itemRequest.id}`] = `${itemRequest.vid + 1}`;
+            }
+        });
+
+        const createOrUpdates = requests.filter((request: BatchReadWriteRequest) => {
+            return request.operation === 'create' || request.operation === 'update';
+        });
+
+        requests.forEach((request: BatchReadWriteRequest) => {
+            if (request.operation === 'create') {
+                idToVersionId[`${request.resourceType}_${request.id}`] = '1';
+            }
+        });
+
+        await Promise.all(
+            createOrUpdates.map(async (request: BatchReadWriteRequest) => {
+                const { resource } = request;
+                const versionedLinksArray = this.versionedLinks[request.resourceType];
+                if (!versionedLinksArray) {
+                    return;
+                }
+                const flattenedResources: Record<string, string> = flatten(resource);
+                const references = Object.keys(flattenedResources).filter((key: string) => {
+                    return key.endsWith('.reference');
+                });
+                if (references.length === 0) {
+                    return;
+                }
+                const versionedLinksSet = new Set(versionedLinksArray);
+                await Promise.all(
+                    references.map(async (referencePath: string) => {
+                        if (!versionedLinksSet.has(referencePath)) {
+                            return;
+                        }
+                        let entryReference = flattenedResources[referencePath];
```

Comment:
nit: more explicit name
```suggestion
                        let entryReferenceValue = flattenedResources[referencePath];
```